iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -1778,13 +1778,19 @@ + (void)registerUserInternal {
 }
 
 + (void)receivedInAppMessageJson:(NSArray<NSDictionary *> *)messagesJson {
-    let messages = [NSMutableArray new];
-
-    for (NSDictionary *messageJson in messagesJson) {
-        let message = [OSInAppMessage instanceWithJson:messageJson];
+    var messages = [NSMutableArray new];
+    
+    if (messagesJson) {
+        for (NSDictionary *messageJson in messagesJson) {
```

Comment:
I'm wondering the difference with android regarding this method because this will affect the redisplay stuff. I will need to check though

Reply:
This def wont affect your redisplay stuff. This is the entry point for IAMs

Reply:
What should we do instead?

Reply:
Handling this now by creating tow methods. On for on_session IAM messages init and one for caching. This splits out the logic for each since different things need to be handled